core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSFileDeleteAction.java

```diff
@@ -60,15 +61,7 @@ public void run(ActionContext context) throws Exception {
     FileSystem fileSystem = path.getFileSystem(new Configuration());
 
     FileStatus[] listFiles;
-    if (config.fileRegex != null) {
-      PathFilter filter = new PathFilter() {
-        private final Pattern pattern = Pattern.compile(config.fileRegex);
-
-        @Override
-        public boolean accept(Path path) {
-          return pattern.matcher(path.getName()).matches();
-        }
-      };
+    if (filter != null) {
```

Comment:
configurePipeline() and run() are called in different contexts. configure is when the pipeline is deployed. run is per run. So setting filter in configure will not make it available here.